sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MethodsShouldNotHaveIdenticalImplementations.cs

```diff
@@ -0,0 +1,126 @@
+        private const string MessageFormat = "Update this method so that its implementation is not identical to '{0}'.";
+
+        private static readonly DiagnosticDescriptor rule =
+            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
+
+        private static readonly ISet<KnownType> ThrowStatementTypesToIgnore = ImmutableHashSet.Create(
+            KnownType.System_NotImplementedException,
+            KnownType.System_NotSupportedException);
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    var classDeclaration = (ClassDeclarationSyntax)c.Node;
+                    var methods = classDeclaration.Members.OfType<MethodDeclarationSyntax>().ToList();
+
+                    var alreadyHandledMethods = new HashSet<MethodDeclarationSyntax>();
+
+                    foreach (var method in methods)
+                    {
+                        if (alreadyHandledMethods.Contains(method))
+                        {
+                            continue;
+                        }
+
+                        var duplicates = methods.Where(m => AreDuplicates(method, m, c.SemanticModel)).ToList();
+                        if (duplicates.Count > 0)
+                        {
+                            alreadyHandledMethods.Add(method);
+                            alreadyHandledMethods.UnionWith(duplicates);
+
+                            foreach (var duplicate in duplicates)
+                            {
+                                c.ReportDiagnostic(Diagnostic.Create(rule, duplicate.Identifier.GetLocation(),
+                                    additionalLocations: new[] { method.Identifier.GetLocation() },
+                                    messageArgs: method.Identifier.ValueText));
+                            }
+                        }
+                    }
+                }, SyntaxKind.ClassDeclaration);
+        }
+
+        private bool AreDuplicates(MethodDeclarationSyntax firstMethod, MethodDeclarationSyntax secondMethod,
+            SemanticModel model)
+        {
+            if (firstMethod == secondMethod)
+            {
+                return false;
+            }
+
+            if (firstMethod.Body != null &&
+                secondMethod.Body != null)
+            {
+                return firstMethod.Body.Statements.Count >= 2 &&
+                    firstMethod.Identifier.ValueText != secondMethod.Identifier.ValueText &&
+                    firstMethod.Body.IsEquivalentTo(secondMethod.Body, false);
+            }
+
+            return false;
+        }
+
+        private bool IsIgnoredThrowStatement(StatementSyntax statement, SemanticModel model)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused private method 'IsIgnoredThrowStatement'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1144)